.github/workflows/main.yml

```diff
@@ -16,11 +16,13 @@ on:
   merge_group:
     types: [checks_requested]
 
+env:
+  NX_CLOUD_ACCESS_TOKEN: ${{ !(github.event_name == 'push' && github.ref != 'main') && secrets.NX_CLOUD_ACCESS_TOKEN || '' }}
```

Comment:
The PR is not really following its description: instead of "disabling Nx Cache" it disables Nx Cloud.
Nx Cloud is not only for caching but divers metrics and developer local build optimization.
The best solution to disable the cache would be to provide the following envvar:
`NX_SKIP_NX_CACHE: ${{ github.event_name != 'push' && startsWith(github.ref, 'release') }}`

Reply:
Actually when you checkout the release branch in local, the `nx build` will take around 30s thanks to the Nx Cloud remote cache (if you have set the NX_CLOUD_ACCES_TOKEN locally with your personal read token).
If the the Nx Cloud is enabled on the CI, even if the NX_SKIP_NX_CAHE is enabled, the CI will send the cache to Nx Cloud which will be used for local build.